### PR TITLE
feat(ts2.0): Migrate to typescript 2.0

### PR DESCRIPTION
**Pull request by nicojs**

### CODE REVIEW

**Comment by nicojs** [comment]
Build error is because we first need to release a new version of stryker-karma-runner. The module integration test is depended on it.

**Comment by simondel** [comment]
So far it looks good to me. I still have to update the README to show the new mutators, or shall we just post them on the website once we release a new Stryker version?

I think it's a good idea to use a proper project to see if we generate the same mutants (or at least get the same score and amount of generated mutants). We could use the Juice Shop by @bkimminich.

**Comment by nicojs** [comment]
Sounds good to me! A sanity check on the juice shop is a good idea. I'll do it tomorrow.

Did the mutators change much? Indeed just update once its merged and released.

Oh yeah, typescript 2 is awesome :)

**Comment by bkimminich** [comment]
Woot! Now that https://github.com/bkimminich/juice-shop is the _almost-official-migration-guinea-pig_ of Stryker, I can offer to trade some Juice Shop stickers for a Stryker-T-Shirt with you guys! :grinning:

**Comment by nicojs** [comment]
@bkimminich Of course you can! And you most certainly deserve a T-shirt  :smile:. Please send me a PM on twitter (@_nicojs) or an email with the details.

File: src/mutators/RemoveConditionalsMutator.ts [hunk 1]
```diff
@@ -1,43 +1,41 @@
...
-  applyMutations(node: ESTree.Node, copy: (obj: any, deep?: boolean) => any): ESTree.Node[] {
-    let nodes: ESTree.Node[] = [];
-
+  applyMutations(node: estree.Node, copy: <T>(obj: T, deep?: boolean) => T): estree.Node[] {
     if (this.canMutate(node)) {
-      let mutatedFalseNode: ESTree.Literal = copy((<ESTree.ConditionalExpression>node).test);
-      this.mutateTestExpression(mutatedFalseNode, false);
-      nodes.push(mutatedFalseNode);
+      let nodes: estree.Node[] = [];
+      nodes.push(this.booleanLiteralNode(node.nodeID, false));
```
Comment (nicojs) [inline src/mutators/RemoveConditionalsMutator.ts]: This is in fact incorrect. We should not replace the current node with a `false`/`true` but the test expression.

**Pull request merged**